src/texture.rs

```diff
@@ -0,0 +1,109 @@
+use std::cell::RefCell;
+use std::collections::hash_map::DefaultHasher;
+use std::collections::BTreeMap;
+use std::hash::Hasher;
+use std::ops::Deref;
+use std::rc::Rc;
+
+use super::{ImStr, ImTextureID};
+
+/// Trait that an object representing a texture usable by the drawing back-end
+/// should implement.
+pub trait ImTexture {
+    /// Get [`ImTextureID`]
+    fn get_id(&self) -> ImTextureID;
+    /// Query texture size, in pixels (for convenience)
+    fn get_size(&self) -> (u32, u32);
+}
+
+/// We implement [`ImTexture`] for [`ImTextureID`] one only to be able to use
+/// it on the ID of the font texture (as is).
+impl ImTexture for ImTextureID {
+    fn get_id(&self) -> ImTextureID { *self }
+    fn get_size(&self) -> (u32, u32) {
+        panic!("Cannot get size of ImTextureID (it is a raw pointer)!")
+    }
+}
+
+/// A handle to a texture.
+///
+/// Wraps a fat pointer to an trait object [`ImTexture`].
+//
+//  `Rc` is necessary to make `Box<ImTexture>` clonable.
+//  We need to clone it to extract its value from a `TextureCache`, an type
+//  internal to imgui-rs, which internally contains a `RefCell`.
+#[derive(Clone)]
+pub struct AnyTexture(Rc<Box<ImTexture>>);
+
+impl AnyTexture {
+    /// Create a new [`AnyTexture`] from an object implementing the [`ImTexture`] trait.
+    fn new<T: 'static + ImTexture>(texture: T) -> Self { AnyTexture(Rc::new(Box::new(texture))) }
+}
+
+impl ImTexture for AnyTexture {
+    fn get_id(&self) -> ImTextureID { self.deref().get_id() }
+    fn get_size(&self) -> (u32, u32) { self.deref().get_size() }
+}
+
+/// Allows to directly use the methods implemnted on [`AnyTexture`]
+impl Deref for AnyTexture {
```

Comment:
Are you sure this is necessary? You've implemented `ImTexture` above anyway for `AnyTexture`, so the deref doesn't seem to add any value here

Reply:
You are right. This is useless! Will get rid of it in the next iteration.